nodes/src/nodes/tool_git/git_repo.py

```
@@ -0,0 +1,980 @@
+            return obj  # type: ignore[return-value]
+        except KeyError:
+            raise GitError(f'Ref {ref!r} not found') from None
+
+    # ------------------------------------------------------------------
+    # Group 1 — Repository
+    # ------------------------------------------------------------------
+
+    def clone(self, url: str, path: str, branch: Optional[str] = None) -> Dict[str, Any]:
+        """Clone *url* into *path*."""
+        if Path(path).exists() and any(Path(path).iterdir()):
+            raise GitError(f'Destination {path!r} exists and is not empty')
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)